Filter specific interface package hierarchy (for reflection)

jerboaa opened the pull request:
The interfaces in javax.naming package are widely implemented across the ecosystem. Registering all implementations of those interfaces for reflection is too much and can actually fail a native build see #51503 where it fails a quickstart. Skip the hierarchical reflection registration for those types to:

1. Reduce the number of classes registered for reflection
2. Reduce the risk of introducing more native build failures

**Testing:**
- [x] Mandrel 23.1 (JDK 21-based) build and run on quarkus with the patch: https://github.com/graalvm/mandrel/actions/runs/20172399899
- [x] Mandrel 25.0 (JDK 25-based) build and run on quarkus with the patch (and `--future-defaults=all`) turned on: https://github.com/graalvm/mandrel/actions/runs/20172317371
- [x] GraalVM CE build of graal/master (LabsJDK-25-based) build and run on quarkus with the patch: https://github.com/graalvm/mandrel/actions/runs/20172192646

Closes: #51503

jerboaa commented:
Native testing seems OK. Marking ready for review.

zakkak reviewed (commented):
I am still looking in the issue and trying to understand what's happening but in the meantime I believe that if ignoring these packages is the going to be the solution it should probably be done by extending https://github.com/quarkusio/quarkus/blob/71538fbca674a6ccbdf5eab57576a9ab946dd975/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveHierarchyBuildItem.java#L360-L363

jerboaa commented:
OK. Thanks.

jerboaa committed:
Filter specific interface package hierarchy

The interfaces in javax.naming package are widely implemented across the
ecosystem. Registering all implementations of those interfaces for
reflection is too much and can actually fail a native build. Skip the
hierarchical reflection registration for those types.

Closes: #51503

jerboaa commented:
> I am still looking in the issue and trying to understand what's happening but in the meantime I believe that if ignoring these packages is the going to be the solution it should probably be done by extending
> 
> https://github.com/quarkusio/quarkus/blob/71538fbca674a6ccbdf5eab57576a9ab946dd975/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveHierarchyBuildItem.java#L360-L363

Fixed in the latest update. I'll re-trigger CI runs linked in the initial comment.

zakkak reviewed (approved):
LGTM, thanks @jerboaa
